Add AppInitializerTest to test ProcessLifecycleOwner

irfano opened the pull request:
This adds a test to check if ProcessLifecycleOwner's observer works correctly. In version `20.1`, the observer was broken, and that caused [AppInitializer#onAppComesFromBackground()](https://github.com/wordpress-mobile/WordPress-Android/blob/737af85f63f2a59bac7ecbc1244030b1e779f58e/WordPress/src/main/java/org/wordpress/android/AppInitializer.kt#L693) not to be called.

To test:
**Success case:**
1. Run `AppInitializerTest`.
2. Ensure `verifyOnAppComesFromBackgroundCalled` passes.

**Fail case:**
1. Break the observer by reverting https://github.com/wordpress-mobile/WordPress-Android/commit/c6f5e2ad6f39a5041607ddfe667740782b4f235a locally.
2. Run `AppInitializerTest`.
3. Ensure `verifyOnAppComesFromBackgroundCalled` fails.

## Regression Notes
1. Potential unintended areas of impact
N/A

4. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

5. What automated tests I added (or what prevented me from doing so)
`AppInitializerTest`

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

irfano committed:
Add AppInitializerTest to test ProcessLifecycleOwner

ParaskP7 reviewed (approved):
👋 @irfano !

I reviewed and tested this PR, everything LGTM! 🌟

Thank you so much for adding this test, it is indeed helpful and will definitely help us identify a lifecycle process problem next time! 🥇 🙇

PS: The only suggestion (💡) I can make here is for you to replace these 2 individual comments you added with one KDoc comment like the following:

```
    /**
     * This tests if [ProcessLifecycleOwner] observer works at the app startup.
     *
     * [WordPress.appIsInTheBackground] is set to false when [ApplicationLifecycleMonitor.onAppComesFromBackground]
     * is called.
     */
```

Note, this KDoc comment will require of you to add a couple of imports as well:

- `import androidx.lifecycle.ProcessLifecycleOwner`
- `import org.wordpress.android.AppInitializer.ApplicationLifecycleMonitor`

No matter, I am approving this PR as my above suggestion is non-blocking. Feel free to apply or just ignore it. 💯

irfano committed:
Update the comment style on AppInitializerTest

irfano committed:
Update assert function on AppInitializerTest

irfano commented:
Thanks for the suggestion, @ParaskP7! Now the comment looks better.
I updated the comment here [fa496e8](https://github.com/wordpress-mobile/WordPress-Android/pull/16980/commits/fa496e83e3da251e627a88b40d7c43d842c585a4).
I also updated `assert` to `assertFalse` for better readability [fdc0b25](https://github.com/wordpress-mobile/WordPress-Android/pull/16980/commits/fdc0b252e313246c1e8ab318b79716abc5cc65f2).

ParaskP7 commented:
This is awesome @irfano ! Double ✅ from my side! ❤️
